sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1982,47 +1970,7 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
     ).foreach { query =>
       assert(intercept[AnalysisException] {
         sql(query)
-      }.getMessage.contains(
-        QueryCompilationErrors.defaultValuesMayNotContainSubQueryExpressions().getMessage))
-    }
-  }
-
-  test("SPARK-39844 Restrict adding DEFAULT columns for existing tables to certain sources") {
```

Comment:
see https://github.com/apache/spark/pull/39942/files#r1102410248

Reply:
Per response to that comment, it looks likely we'll have to retain this functionality.